### PR TITLE
[AIR/Docs] Clarify Dataset configurations

### PR DESCRIPTION
**Pull request by amogkam**
Signed-off-by: Amog Kamsetty <amogkamsetty@yahoo.com>

Clarify docs and error messages for dataset configurations based on user feedback.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-air/check-ingest.rst [hunk 1]
```diff
@@ -23,15 +23,52 @@ user-defined function to preprocess batches of data, and (3) runs an AIR Trainer
...
+    :language: python
+    :start-after: __config_4__
+    :end-before: __config_4_end__
+
+.. _air-configure-ingest:
+
+Configuring Ingest
+------------------
+You can use the :py:class:`~ray.air.config.DatasetConfig` object to configure how Datasets are preprocessed and split across training workers.
+Each :py:class:`~ray.train.data_parallel_trainer.DataParallelTrainer` has a default ``_dataset_config`` class field. It is a mapping
```
Comment (richardliaw) [inline doc/source/ray-air/check-ingest.rst:49]: maybe no need to highlight this private implementation?

Comment (richardliaw) [reply doc/source/ray-air/check-ingest.rst:49]: I think you can just write about how to pass it in

File: python/ray/train/base_trainer.py [hunk 2]
```diff
@@ -214,14 +214,25 @@ def _validate_attributes(self):
...
-        ):
-            raise ValueError(
-                f"At least one value in the `datasets` dict is not a "
-                f"`ray.data.Dataset`: {self.datasets}"
-            )
+        else:
+            for key, dataset in self.datasets.items():
+                if isinstance(dataset, ray.data.DatasetPipeline):
+                    raise ValueError(
+                        f"The Dataset under {key} key is a `ray.data.DatasetPipeline`."
```
Comment (richardliaw) [inline python/ray/train/base_trainer.py:221]:
```suggestion
                        f"The Dataset under '{key}' key is a `ray.data.DatasetPipeline`."
```

**Review by richardliaw — CHANGES_REQUESTED** [review]
nice! one comment on docs

**Comment by amogkam** [comment]
@richardliaw updated, PTAL!

**Review by richardliaw — APPROVED** [review]
great! thanks for the fast turnaround

**Pull request merged**